lib/bridge/IrcBridge.js

```diff
@@ -754,6 +769,128 @@ IrcBridge.prototype.getBotClient = function(server) {
     });
 }
 
+
+var _solveTimeout = null;
+
+IrcBridge.prototype._updateVisibilityMap = function(isMode, key, value) {
+    let hasChanged = false;
+    if (isMode) {
+        if (typeof value !== 'boolean') {
+            throw new Error('+s state must be indicated with a boolean');
+        }
+        if (this._visibilityMap.channelModes[key] !== value) {
+            this._visibilityMap.channelModes[key] = value;
+            hasChanged = true;
+        }
+    }
+    else {
+        if (typeof value !== 'string' || (value !== "private" && value !== "public")) {
+            throw new Error('Room visibility must = "private" | "public"');
+        }
+
+        if (this._visibilityMap.roomVisibilities[key] !== value) {
+            this._visibilityMap.roomVisibilities[key] = value;
+            hasChanged = true;
+        }
+    }
+
+    if (hasChanged) {
+        clearTimeout(_solveTimeout);
+
+        _solveTimeout = setTimeout(() => {
+            this._solveVisibility()
+        }, 10000);
```

Comment:
What is the purpose of maintaining a timer here? I'm guessing this is to avoid the risk of the bridge wedging entirely on large graphs (or maybe we aren't doing cycle detection so could loop forevaaaar?). A comment to state as much would be good, and given you're returning a Promise in this function, I would be tempted to use `Promise.timeout()` rather than wrangling timeout IDs and `setTimeout` yourself.